### ISSUE TITLE
SC2126: `grep --quiet` over `[ $(grep -c) -gt 0 ]` etc

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3295]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2126
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# [ "$(grep foo file | wc -l)" -gt 0 ]  # SC2126
[ "$(grep -c foo file)" -ge 1 ]
[ "$(grep -c foo file)" -gt 0 ]
[ "$(grep -c foo file)" -ne 0 ]
```

#### Here's what shellcheck currently says:

nothing

#### Here's what I wanted or expected to see:

```
[ "$(grep -c foo file)" -ge 1 ]
^-----------------------------^ SCXXXX (style): Consider using 'grep -q'
[ "$(grep -c foo file)" -gt 0 ]
^-----------------------------^ SCXXXX (style): Consider using 'grep -q'
[ "$(grep -c foo file)" -ne 0 ]
^-----------------------------^ SCXXXX (style): Consider using 'grep -q'

```

Actually this is mentioned in <https://github.com/koalaman/shellcheck/wiki/SC2126> but is not addressed.
